src/storage/cuckoo/cuckoo.c

```diff
@@ -100,12 +100,12 @@ _select_candidate(const uint32_t offset[])
     if (cuckoo_policy == CUCKOO_POLICY_RANDOM) {
         selected = offset[RANDOM(D)];
     } else if (cuckoo_policy == CUCKOO_POLICY_EXPIRE) {
-        proc_time_i expire, min = UINT32_MAX; /* legal ts should < UINT32_MAX */
+        proc_time_i expire, min = INT32_MAX; /* legal ts should <= INT32_MAX */
         uint32_t i;
 
         for (i = 0; i < D; ++i) {
             expire = item_expire(OFFSET2ITEM(offset[i]));
-            if (expire < min) {
+            if (expire <= min) {
```

Comment:
if `ttl = 0` gets translated to `INT32_MAX`, we can assume that any expiry less than that is an actual expiry instead of an approximation for "never expire". Since we start by "selecting" the first offset, in the case where no item has a shorter TTL, we will still go with the first offset, regardless of the value chosen to indicate max TTL. So my understanding is you can keep the current code unchanged and it will still work with the new init value of `min`, and it will retain the behavior of evicting the first item if there's nothing expiring sooner, is that correct?

Reply:
hm, I think you are correct. we don't need to make an exception for unbounded timestamp, i think i was thrown off by the "legal timestamp" comment. cool, i'll change this back to the original behavior.